integrationTests/testInitializer.go

```diff
@@ -622,3 +647,497 @@ func extractUint64ValueFromTxHandler(txHandler data.TransactionHandler) uint64 {
+	hashes := make([][]byte, 0)
+
+	for _, miniblock := range body {
+		for _, shardId := range shardIds {
+			if miniblock.ReceiverShardID == shardId {
+				buff, _ := TestMarshalizer.Marshal(miniblock)
+				hashes = append(hashes, TestHasher.Compute(string(buff)))
+			}
+		}
+	}
+
+	return hashes
+}
+
+// EqualSlices checks if two slices are equal
+func EqualSlices(slice1 [][]byte, slice2 [][]byte) bool {
+	if len(slice1) != len(slice2) {
+		return false
+	}
+
+	//check slice1 has all elements in slice2
+	for _, buff1 := range slice1 {
+		found := false
+		for _, buff2 := range slice2 {
+			if bytes.Equal(buff1, buff2) {
+				found = true
+				break
+			}
+		}
+		if !found {
+			return false
+		}
+	}
+
+	//check slice2 has all elements in slice1
+	for _, buff2 := range slice2 {
+		found := false
+		for _, buff1 := range slice1 {
+			if bytes.Equal(buff1, buff2) {
+				found = true
+				break
+			}
+		}
+		if !found {
+			return false
+		}
+	}
+	return true
+}
+
+// GeneratePrivateKeyInShardId generates a private key that is in a given shard
+func GeneratePrivateKeyInShardId(
+	coordinator sharding.Coordinator,
+	shardId uint32,
+) crypto.PrivateKey {
+
+	suite := kyber.NewBlakeSHA256Ed25519()
+	keyGen := signing.NewKeyGenerator(suite)
+	sk, pk := keyGen.GeneratePair()
+
+	for {
+		buff, _ := pk.ToByteArray()
+		addr, _ := TestAddressConverter.CreateAddressFromPublicKeyBytes(buff)
+
+		if coordinator.ComputeId(addr) == shardId {
+			return sk
+		}
+
+		sk, pk = keyGen.GeneratePair()
+	}
+}
+
+// GenerateRandomHexAddressInShard generates a random address in the given shard
+func GenerateRandomHexAddressInShard(
+	coordinator sharding.Coordinator,
+	addrConv state.AddressConverter,
+) string {
+
+	addrBytes := []byte(CreateRandomHexString(32))
+	for {
+		addr, _ := addrConv.CreateAddressFromPublicKeyBytes(addrBytes)
+		if coordinator.ComputeId(addr) == coordinator.SelfId() {
+			return hex.EncodeToString(addrBytes)
+		}
+		addrBytes = []byte(CreateRandomHexString(32))
+	}
+}
+
+// CreateMintingForSenders creates account with balances for every node in a given shard
+func CreateMintingForSenders(
+	nodes []*TestProcessorNode,
+	senderShard uint32,
+	sendersPrivateKeys []crypto.PrivateKey,
+	value *big.Int,
+) {
+
+	for _, n := range nodes {
+		//only sender shard nodes will be minted
+		if n.ShardCoordinator.SelfId() != senderShard {
+			continue
+		}
+
+		for _, sk := range sendersPrivateKeys {
+			pkBuff, _ := sk.GeneratePublic().ToByteArray()
+			adr, _ := TestAddressConverter.CreateAddressFromPublicKeyBytes(pkBuff)
+			account, _ := n.AccntState.GetAccountWithJournal(adr)
+			_ = account.(*state.Account).SetBalanceWithJournal(value)
+		}
+
+		_, _ = n.AccntState.Commit()
+	}
+}
+
+// GenerateDefaultHeaderAndBody creates a default header and body
+func GenerateDefaultHeaderAndBody(senderShard uint32, recvShards ...uint32) (data.BodyHandler, data.HeaderHandler) {
```

Comment:
do we need another function that creates a body and a header? we already have ProposeBlock

Reply:
No. Removed.